static/js/whyis_vue/components/add-link.vue

```diff
@@ -140,12 +169,22 @@ export default Vue.component('add-link', {
             this.entityList = this.getNeighborEntities(this.uri);
         },
         resolveEntity(query){
-            if (!query.label) {
-                if (query.length > 2) {
-                    this.entityList = this.getEntityList(query);
-                } else
-                    this.entityList = this.getNeighborEntities(this.uri);
+            var thisVue = this;
+            this.entityQuery = query
+            // Debounce for entity search 
+            if (!thisVue.awaitingEntity) {
+                setTimeout(function () {
+                    let entityQuery = thisVue.entityQuery;
+                    if (!entityQuery.label) {
+                        if (entityQuery.length > 2) {
+                            thisVue.entityList = thisVue.getEntityList(entityQuery);
+                        } else
+                            thisVue.entityList = thisVue.getNeighborEntities(thisVue.uri);
+                    }
+                    thisVue.awaitingEntity = false;
+                }, 1000); 
```

Comment:
For future reference, we have a custom debounce method under `utilities/`.  It has proven to work with methods that access the component scope without any extra hassle.